### PR TITLE
Implementation of Context Menus

### PR DESCRIPTION
**Pull request by danwalmsley**

### CODE REVIEW

File: src/Perspex.Controls/ContextMenu.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+        /// <summary>
+        /// The popup window used to display the active context menu.
+        /// </summary>
+        private static PopupRoot s_popup;
+
+        /// <summary>
+        /// The control that the currently visible context menu is attached to.
+        /// </summary>
+        private static Control s_current;
```
Comment (grokys) [inline src/Perspex.Controls/ContextMenu.cs]: Is `s_current` actually needed? It's assigned but never used.

File: src/Perspex.Controls/ContextMenu.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    public class ContextMenu : SelectingItemsControl
+    {
+        private bool _isOpen;
+
+        /// <summary>
+        /// Defines the ContextMenu.Menu attached property.
+        /// </summary>
+        //public static readonly AttachedProperty<object> ContextMenuProperty =
+        //    PerspexProperty.RegisterAttached<ContextMenu, Control, object>("Menu");
```
Comment (grokys) [inline src/Perspex.Controls/ContextMenu.cs]: Probably just remove this for now. I don't like comment out code in the repos ;) We can add it back later when OmniXAML is fixed.

File: src/Perspex.Controls/ContextMenu.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        /// Gets the value of the ToolTip.Tip attached property.
+        /// </summary>
+        /// <param name="element">The control to get the property from.</param>
+        /// <returns>
+        /// The content to be displayed in the control's tooltip.
+        /// </returns>
+        public static object GetContextMenu(Control element)
+        {
+            return element.GetValue(ContextMenuProperty);
+        }
```
Comment (grokys) [inline src/Perspex.Controls/ContextMenu.cs]: Same with these.

**Comment by danwalmsley** [comment]
addressed the issues raise.

**Pull request merged**